### PR TITLE
[fixit] Add library versions to docs

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

Shows library versions in addition to core version in docs.

- If core version is >= 1.0, then version dropdown will show e.g. "1.0.9 /0.16.9 (libs)"
- If core version is < 1.0, only one version is shown
- Changed display of `(latest)` for latest version to drop the parentheses and use a lavender background. This was necessary because of the new `(libs)` text.

### How I Tested These Changes

Browsed dev site locally.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9768** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9768" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9768?utm_source=stack-comment).

**Review by yuhan — COMMENTED** [review]
looking good. should we change the ones in the dropdown as well?
image

File: docs/next/components/VersionDropdown.tsx [hunk 1]
```diff
@@ -6,23 +6,38 @@ import {useVersion} from '../util/useVersion';
...
                     <span className="flex min-w-0 items-center justify-between space-x-3">
-                      <span className="flex-1 min-w-0">
-                        <span className="text-gray-900 dark:text-gray-300 text-xs lg:text-sm truncate">
-                          {currentVersion} {currentVersion === latestVersion && '(latest)'}
+                      <span className="flex-1 min-w-0 text-gray-900 dark:text-gray-300 text-xs lg:text-sm truncate space-x-1">
+                        <span>
+                          {currentVersion} {libraryVersionText}
                         </span>
+                        {currentVersion === latestVersion ? (
+                          <span className="bg-lavender rounded-full px-2 py-1">latest</span>
```
Comment (erinkcochran87) [inline docs/next/components/VersionDropdown.tsx:39]: 🔥

File: docs/next/components/VersionDropdown.tsx [hunk 2]
```diff
@@ -6,23 +6,38 @@ import {useVersion} from '../util/useVersion';
...
+function getLibraryVersion(coreVersion) {
+  const [major, minor, patch] = coreVersion.split('.');
+  return parseInt(major) < 1 ? coreVersion : ['0', 16 + parseInt(minor), patch].join('.');
+}
+
+function getLibraryVersionText(coreVersion) {
+  const libraryVersion = coreVersion === 'master' ? 'master' : getLibraryVersion(coreVersion);
+  return libraryVersion === 'master' || coreVersion === libraryVersion
+    ? ''
+    : `/ ${libraryVersion} (libs)`;
```
Comment (erinkcochran87) [inline docs/next/components/VersionDropdown.tsx:18]: Is it safe to assume users will know what `libs` means?

File: docs/next/components/VersionDropdown.tsx [hunk 3]
```diff
@@ -59,12 +74,16 @@ export default function VersionDropdown() {
                       <span className="text-sm font-medium leading-5 text-gray-900">
                         {currentVersion}
                       </span>
-                      . You can select a different version below.
+                      . You can select a different version below. Note that prior to 1.0, all
```
Comment (erinkcochran87) [inline docs/next/components/VersionDropdown.tsx:77]: A few thoughts here:

- It feels a bit odd to have this copy display in its entirety for versions prior to 1.0 - to me it seems unnecessary
- It's also a bit long for its current location. Perhaps we could have an icon with a tooltip (or something) display when a version with diverging lib versions is selected
   We could link out to the page that explains our versioning methodology, too: `/getting-started/releases#dagster-integration-libraries`

**Review by erinkcochran87 — COMMENTED** [review]
Thanks for taking this on, Sean! I think this could use a bit more refining, but overall it's a good step in the right direction.

**Comment by yuhan** [comment]
looks look backend wise. defer to erin for how we present the info

File: docs/next/components/VersionDropdown.tsx (hunk 3 above)
Comment (smackesey) [reply docs/next/components/VersionDropdown.tsx:77]: - The copy only displays when you click the menu, which is the same regardless of the version you're browsing, right? It's meant to explain why some entries show two versions and some show one version. If you are on an older version and browsing the list of all versions (including the new ones), you want to know this information.
- I agree it's a little long, but I don't see how to compress it while still explaining what must be explained. I attempted a tooltip on hover over the dropdown, but could not find a pre-existing tooltip component in the docs site (or Headless UI framework) and didn't have the time to spend creating a custom one now (I don't think there are tooltips elsewhere in docs).

File: docs/next/components/VersionDropdown.tsx (hunk 2 above)
Comment (smackesey) [reply docs/next/components/VersionDropdown.tsx:18]: No, but that's what the explanatory text in the dropdown is for.

**Review by erinkcochran87 — APPROVED** [review]

**Pull request merged**